<launch>
  
  <arg name="input_topic" default="/input"/>
  <arg name="output_topic" default="/output"/>
  <arg name="common_frame_id" default="world"/>
  <arg name="output_resolution" default="0.01"/>
  <arg name="termination_method" default="time_window"/>
  <arg name="termination_value" default="500"/>
  <!-- number of times to allow inconsistent time window before resizing (like inverse of an adaptation rate) -->
  <arg name="time_window_inconsistency_thresh" default="0"/>

  <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen"/>
  <node pkg="nodelet" type="nodelet" name="mixed_concatenate_server_nodelet" args="load pcl_ros_toolbox/MixedConcatenateServer pcl_manager" output="screen">
    <remap from="~input" to="$(arg input_topic)" />
    <remap from="~output" to="/concatenated_cloud" />
		<param name="termination_method" value="$(arg termination_method)"/>
		<param name="termination_value" value="$(arg termination_value)"/>
		<param name="common_frame_id" value="$(arg common_frame_id)"/>
    <param name="time_window_inconsistency_thresh" value="$(arg time_window_inconsistency_thresh)"/>
	</node>
  <node pkg="nodelet" type="nodelet" name="voxel_grid_filter_nodelet" args="load pcl/VoxelGrid pcl_manager" output="screen">
        <remap from="~input" to="/concatenated_cloud" />
        <remap from="~output" to="$(arg output_topic)" />
        <param name="filter_field_name" value="z"/>
        <param name="filter_limit_min" value="-100000"/>
        <param name="filter_limit_max" value="100000"/>
        <param name="filter_limit_negative" value="false"/>
        <param name="leaf_size" value="$(arg output_resolution)"/>
	</node>

</launch>